includes/class-wc-payments-db.php

```diff
@@ -87,17 +87,22 @@ public function order_from_intent_id( $intent_id ) {
 	 * @return null|string
 	 */
 	private function order_id_from_meta_key_value( $meta_key, $meta_value ) {
-		global $wpdb;
-
-		// The order ID is saved to DB in `WC_Payment_Gateway_WCPay::process_payment()`.
-		$order_id = $wpdb->get_var(
-			$wpdb->prepare(
-				"SELECT DISTINCT MAX(ID) FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_key = %s AND meta.meta_value = %s",
-				$meta_key,
-				$meta_value
-			)
+		$orders = wc_get_orders(
+			[
+				'limit'      => 1,
+				'meta_key'   => $meta_key, //phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				'meta_value' => $meta_value, //phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+			]
 		);
-		return $order_id;
+		if ( $orders && ! empty( $orders ) ) {
+			/**
+			 * As wc_get_orders may also return stdClass, Psalm infers error.
+			 *
+			 * @psalm-suppress UndefinedMethod
+			 */
+			return (string) $orders[0]->get_id();
```

Comment:
Not directly related to this PR but in some places in the codebase, I have seen the functions that return ids from DB are using the `int|false` signature whereas in some it is `string|null`, just curious which is the recommended approach to follow.

Reply:
`string` is used here to maintain compatibility with the existing function signature. I feel `<something>|null` is preferable over `<something>|false`, because the latter is changing the return type.

Reply:
I get that the existing signature is maintained, my question was more general about how we use different signatures for similar things across the codebase, but unrelated to this so resolving this for now.